org.eclipse.lsp4e/src/org/eclipse/lsp4e/outline/SymbolsLabelProvider.java

```diff
@@ -335,4 +350,11 @@ public void preferenceChange(PreferenceChangeEvent event) {
 		}
 	}
 
+	private static String getUri(WorkspaceSymbol symbol) {
```

Comment:
This can just be
`return symbol.getLocation().map(Location::getUri, WorkspaceSymbolLocation::getUri)`

Reply:
Ooh, nice, I was trying to figure out how `Either.map(..)` was supposed to be used! Thanks.

I thought of moving the code back inline (since you've shown it can be indeed done in one line), but I think it still reads better out of line.